src/OhioBox.Moranbernate/Utils/IKeyValuePairBuilder.cs

```diff
@@ -8,22 +8,74 @@ namespace OhioBox.Moranbernate.Utils
 	public interface IKeyValuePairBuilder<T>
 	{
 		IKeyValuePairBuilder<T> Set<TValue>(Expression<Func<T, TValue>> expression, TValue value);
+		IKeyValuePairBuilder<T> Increment<TValue>(Expression<Func<T, TValue>> expression, TValue value);
 	}
 
 	internal class KeyValuePairBuilder<T> : IKeyValuePairBuilder<T>
 	{
-		private readonly List<Tuple<Property, object>> _list = new List<Tuple<Property, object>>();
-
+		private readonly List<IUpdatedProperty> _list = new List<IUpdatedProperty>();
+		
 		public IKeyValuePairBuilder<T> Set<TValue>(Expression<Func<T, TValue>> expression, TValue value)
 		{
-			var tuple = Tuple.Create(ExpressionProcessor<T>.GetPropertyFromCache(expression), (object)value);
-			_list.Add(tuple);
+			var settableProperty = new ValueUpdatedProperty(ExpressionProcessor<T>.GetPropertyFromCache(expression), value);
+			_list.Add(settableProperty);
 			return this;
 		}
 
-		public IEnumerable<Tuple<Property, object>> GetEnumerable()
+		public IKeyValuePairBuilder<T> Increment<TValue>(Expression<Func<T, TValue>> expression, TValue value)
+		{
+			var incrementableProperty = new ValueIncrementedProperty(ExpressionProcessor<T>.GetPropertyFromCache(expression), value);
+			_list.Add(incrementableProperty);
+
+			return this;
+		}
+
+		public IList<IUpdatedProperty> GetEnumerable() => _list;
+	}
+
+	internal interface IUpdatedProperty
+	{
+		Property Property { get; }

```

Comment:
if i understand this interface represents the property to update, if so qwhy not just call it like that - IPropertyToUpdate and PropertyToIncrement etc.

Reply:
I personally don't love XXXToSomthing syntax but I get your point